apollo-android-support/src/androidTest/java/com/apollographql/apollo/cache/normalized/sql/ApolloPrefetchCallbackTest.java

```diff
@@ -0,0 +1,175 @@
+package com.apollographql.apollo.cache.normalized.sql;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.os.Message;
+import android.support.test.runner.AndroidJUnit4;
+
+import com.apollographql.apollo.ApolloCall;
+import com.apollographql.apollo.ApolloPrefetch;
+import com.apollographql.apollo.ApolloPrefetchCallback;
+import com.apollographql.apollo.ApolloClient;
+import com.apollographql.apollo.ApolloPrefetchCallback;
+import com.apollographql.apollo.api.Query;
+import com.apollographql.apollo.api.Response;
+import com.apollographql.apollo.api.ResponseFieldMapper;
+import com.apollographql.apollo.api.ResponseReader;
+import com.apollographql.apollo.exception.ApolloException;
+import com.apollographql.apollo.exception.ApolloHttpException;
+import com.apollographql.apollo.exception.ApolloNetworkException;
+import com.apollographql.apollo.exception.ApolloParseException;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.annotation.Nonnull;
+
+import okhttp3.OkHttpClient;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+import static com.google.common.truth.Truth.assertThat;
+import static junit.framework.Assert.fail;
+
+@RunWith(AndroidJUnit4.class)
+public class ApolloPrefetchCallbackTest {
+  @Rule public final MockWebServer server = new MockWebServer();
+  private ApolloClient apolloClient;
+  private Query emptyQuery;
+
+  @Before public void setUp() throws Exception {
+    apolloClient = ApolloClient.builder()
+        .serverUrl(server.url("/"))
+        .okHttpClient(new OkHttpClient.Builder()
+            .connectTimeout(2, TimeUnit.SECONDS)
+            .readTimeout(2, TimeUnit.SECONDS)
+            .build())
+        .build();
+
+    emptyQuery = new Query() {
```

Comment:
seems like multiple tests use this, can we make a factory to dedup the code? Alternatively put in `TestUilts` or smth